src/realm/util/encrypted_file_mapping.cpp

```diff
@@ -233,7 +234,11 @@ bool AESCryptor::read(int fd, off_t pos, char* dst, size_t size)
             }
         }
 
-        crypt(mode_Decrypt, pos, dst, m_rw_buffer.get(), reinterpret_cast<const char*>(&iv.iv1));
+        // decrypt to a temporary buffer first because some implementations may
+        // require several iterations on a block before producing a final result
+        // and our shared mappings might be read by other readers at any time
+        crypt(mode_Decrypt, pos, m_dst_buffer.get(), m_rw_buffer.get(), reinterpret_cast<const char*>(&iv.iv1));
```

Comment:
I couldn't understand this... how about something along the lines of:

Some address ranges of the destination buffer of AESCryptor::read() may be expected to stay unmodified, i.e. not being touched and may have read-access from other threads while decrypting is taking place.

However, some implementations of AES_cbc_encrypt() (in particular OpenSSL) will intermediately put garbled bytes at the destination during operation which will lead to crashes.

We therefore decrypt to a temporary buffer which we memcpy() at the end.